File: application/src/test/java/com/smtm/application/security/AuthenticationStepdefs.java
package com.smtm.application.security;

import java.util.Map;
import com.smtm.application.security.v1.CredentialsDto;
import io.cucumber.java.DataTableType;
import io.cucumber.java.en.Given;

public class AuthenticationStepdefs {

    private final FakeAuthentication fakeAuthentication;

    public AuthenticationStepdefs(FakeAuthentication fakeAuthentication) {
        this.fakeAuthentication = fakeAuthentication;
    }

    @Given("credentials produces a valid token")
    public void credentialsProducesAValidToken(CredentialsDto credentialsDto) {
        fakeAuthentication.setValidTokenFor(credentialsDto);
    }

    @DataTableType
    public CredentialsDto credentialsDto(Map<String, String> credentials) {
        return CredentialsDto.of(credentials.get("email"), credentials.get("password"));
    }
}

File: application/src/test/kotlin/com/smtm/application/security/AuthenticationStepdefs.kt
package com.smtm.application.security

import com.smtm.application.security.v1.CredentialsDto
import com.smtm.application.security.v1.credentialsDtoOf
import com.smtm.security.token.tokenOf
import io.cucumber.java.DataTableType
import io.cucumber.java.en.Given

class AuthenticationStepdefs(private val fakeAuthentication: FakeAuthentication) {

    @Given("token {string} is produced by credentials")
    fun tokenIsProducedByCredentials(token: String, credentials: CredentialsDto) {
        fakeAuthentication.validTokens[credentials] = tokenOf(token, 1)
    }

    @DataTableType
    fun credentialsDto(credentials: Map<String, String>): CredentialsDto {
        return credentialsDtoOf(credentials.getValue("email"), credentials.getValue("password"))
    }
}
